internal/services/search/validate/search_service_partition_count.go

```diff
@@ -0,0 +1,20 @@
+package validate
+
+import (
+	"fmt"
+)
+
+func PartitionCount(v interface{}, k string) (warnings []string, errors []error) {
+	partitionCount := v.(int)
+
+	switch partitionCount {
+	case 5, 7, 8, 9, 10, 11:
+		errors = append(errors, fmt.Errorf("%q must be 1, 2, 3, 4, 6, or 12, got %d", k, partitionCount))
+	}
+
+	if partitionCount > 12 {
+		errors = append(errors, fmt.Errorf("%q must be 1, 2, 3, 4, 6, or 12, got %d", k, partitionCount))
+	}
+
+	return warnings, errors
```

Comment:
we can reuse an existing validation function here:

```suggestion
	return validation.IntInSlice([]int{
		1,
		2,
		3,
		4,
		6,
		12,
	})
```

Reply:
Fixed.